### PR TITLE
Mirego-base-setup 7 of X: Killswitch

### PR DESCRIPTION
**Pull request by ChristopheTremblay**
## 📖 Description
killswitch integration

## 📝 Notes
The apiKey the Const file is a temporary string `Replace with your own killswitchApiKey`

Simulator Screenshot - iPhone 15 Pro - 2023-11-06 at 11 37 51

Screenshot 2023-11-06 at 11 38 02

### CODE REVIEW

File: shared/src/commonMain/kotlin/com/mirego/kmp/boilerplate/bootstrap/AppEnvironment.kt [hunk 1]
```diff
@@ -2,14 +2,34 @@ package com.mirego.kmp.boilerplate.bootstrap
...
+            killSwitchAPIKey = KILLSWITCH_API_KEY
+        )
     )
 }
+
+data class PlatformSpecific(
+    val killSwitchAPIKey: String
+)
+
+private const val KILLSWITCH_API_KEY = "809b16d21b5c9525b10f7298ad90b1ce3a4c27223f674d5ac7a2b6fc060929d0"
```
Comment (remi) [inline shared/src/commonMain/kotlin/com/mirego/kmp/boilerplate/bootstrap/AppEnvironment.kt:35]: We should refrain from using a real value here.

Comment (ChristopheTremblay) [reply shared/src/commonMain/kotlin/com/mirego/kmp/boilerplate/bootstrap/AppEnvironment.kt:35]: Ah yeah indeed my bad, c'était pour m'assurer que ça fonctionne mais ça devrait être une string TODO or something

File: androidApp/src/main/java/com/mirego/kmp/boilerplate/app/MainActivity.kt [hunk 2]
```diff
@@ -20,12 +29,28 @@ class MainActivity : AppCompatActivity() {
...
         installSplashScreen()
         WindowCompat.setDecorFitsSystemWindows(window, false)
         setContent {
             ApplicationView(applicationViewModel = viewModel)
         }
+
+        lifecycleScope.launch {
+            try {
+                AndroidKillswitch.showDialog(
+                    AndroidKillswitch.engage(appEnvironment.androidSpecific.killSwitchAPIKey, this@MainActivity, Const.KILLSWITCH_URL), // TODO use current appEnvironment
```
Comment (mathieularue) [inline androidApp/src/main/java/com/mirego/kmp/boilerplate/app/MainActivity.kt:47]: Is the TODO still relevant?

**Review by mathieularue — APPROVED** [review]

File: shared/src/commonMain/kotlin/com/mirego/kmp/boilerplate/bootstrap/AppEnvironment.kt (hunk 1 above)
Comment (OlivierPineau) [reply shared/src/commonMain/kotlin/com/mirego/kmp/boilerplate/bootstrap/AppEnvironment.kt]: It crashes when the key is empty.
The killswitch library should handle that case

**Review by sdetilly — COMMENTED** [review]
It seems you are also adding analytics here

**Pull request merged**